src/main/java/filter/lexer/Lexer.java

```diff
@@ -21,6 +21,7 @@
             // These have higher priority than Symbol
             new Rule("\\d{4}-\\d{1,2}-\\d{1,2}", TokenType.DATE), // YYYY-MM?-DD?
             new Rule("[A-Za-z]+(-[A-Za-z]+)*\\s*:", TokenType.QUALIFIER),
+            new Rule("[A-Za-z0-9#][/A-Za-z0-9.'+-]+#", TokenType.COMPOUND_ID),
```

Comment:
So, the regex should check that it fits syntax of a repo id. We get the range support for free, so I think we can do it this way. The range thing needs tests.

Maybe the token type could be named `COMPOUND_ID_PREFIX`?